### PR TITLE
Remove deprecated code from StoreSinking

### PR DESCRIPTION
**Pull request by 0xdaryl**
A previous cleanup of this code left some unreachable remnants behind.  Remove them now.

Signed-off-by: Daryl Maier <maier@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@genie-omr build all

File: compiler/optimizer/SinkStores.cpp [hunk 1]
```diff
@@ -619,19 +613,6 @@ int32_t TR_SinkStores::performStoreSinking()
    _placementsForEdgesToBlock = (TR_EdgeStorePlacementList **) trMemory()->allocateStackMemory(arraySize);
    memset(_placementsForEdgesToBlock, 0, arraySize);
 
-   // some extra meta data is needed if stores with indirect loads are going to be sunk
-   if (sinkStoresWithIndirectLoads())
```
Comment (0xdaryl) [inline compiler/optimizer/SinkStores.cpp:623]: Note to reviewers:  this function always returned false, hence the code it guards was removed.

File: compiler/optimizer/SinkStores.cpp [hunk 2]
```diff
@@ -2320,250 +2261,6 @@ TR_SinkStores::replaceLoadsWithTempSym(TR::Node *newNode, TR::Node *origNode, TR
    }
 
 
-// A single store that is being replicated to a side exit may have 0 or more nodes replaced with anchored nodes for different reasons
-// insertAnchoredNodes will take care of all possible node replacements so only a single recursive walk of a store is required
-uint32_t
-TR_SinkStores::insertAnchoredNodes(TR_MovableStore *store,
```
Comment (0xdaryl) [inline compiler/optimizer/SinkStores.cpp:2326]: Note to reviewers:  this function is unreachable (except via a recursive call).  It is removed as well as the data structures that only it used.  Most of the other deletions in this PR are derived from this removal.

**Comment by 0xdaryl** [comment]
@vijaysun-omr : you asked me privately to add a comment to the code mentioning that handling of indirects was removed in a previous PR.  I didn't find a convenient place to mention that in the code, so instead I amended the commit message itself to reference PR #3893 that led to these vestigial elements being cleaned up here.

**Comment by 0xdaryl** [comment]
Just a reminder ping @vijaysun-omr .  No urgency on this though.

**Comment by vijaysun-omr** [comment]
Thanks, I will merge once tests pass

**Comment by vijaysun-omr** [comment]
@genie-omr build all

**Comment by vijaysun-omr** [comment]
@genie-omr build all

**Comment by vijaysun-omr** [comment]
@genie-omr build zos

**Comment by vijaysun-omr** [comment]
This change is not platform dependent as such and the zOS build machine issues have been persisting for a few days now. So I am merging this change (rather than wait for the zOS issues to be resolved).

**Pull request merged**